File: Assemblies/Protean.Tools.vb/Integration/Twitter/TwitterVB2/Objects/ATwiCliPhotoFile.vb
Namespace Integration.Twitter.TwitterVB2
    Public Class ATwiCliPhotoFile
        Inherits ATwiCliFile

#Region "TCPF Constants"
        Const TCPF_PHOTO_STRING As String = "photo.jpg"
        Const TCPF_PHOTO_EXTENSION As String = ".jpg"
#End Region
        Private m_iUserTags As Integer
        Private m_dLongitude As Double
        Private m_dLatitude As Double
        Private m_bShowMap As Boolean
        Private m_strCameraMake As String
        Private m_strCameraModel As String
        Public Property UserTags_Count() As Integer
            Get
                Return m_iUserTags

            End Get
            Set(ByVal value As Integer)
                m_iUserTags = value

            End Set
        End Property

        Public Property Longitude() As Double
            Get
                Return m_dLongitude

            End Get
            Set(ByVal value As Double)
                m_dLongitude = value
            End Set
        End Property

        Public Property Latitude() As Double
            Get
                Return m_dLatitude

            End Get
            Set(ByVal value As Double)
                m_dLatitude = value

            End Set
        End Property

        Public Property ShowMap() As Boolean
            Get
                Return m_bShowMap

            End Get
            Set(ByVal value As Boolean)
                m_bShowMap = value

            End Set
        End Property

        Public Property CameraMake() As String
            Get
                Return m_strCameraMake

            End Get
            Set(ByVal value As String)
                m_strCameraMake = value

            End Set
        End Property

        Public Property CameraModel() As String
            Get
                Return m_strCameraModel

            End Get
            Set(ByVal value As String)
                m_strCameraModel = value

            End Set
        End Property

        Public Overrides Function GetDirectURL() As String
            GetDirectURL = TCF_DIRECT_URL & TCPF_PHOTO_STRING & "?id=" & Me.ID & "&size=original"
        End Function

        Public Overrides Function GetExtension() As String
            Return TCPF_PHOTO_EXTENSION

        End Function

        Public ReadOnly Property LargeURL() As String
            Get
                Return TCF_DIRECT_URL & TCPF_PHOTO_STRING & "?id=" & Me.ID & "&size=large"
            End Get
        End Property

        Public ReadOnly Property SmallURL() As String
            Get
                Return TCF_DIRECT_URL & TCPF_PHOTO_STRING & "?id=" & Me.ID & "&size=small"
            End Get
        End Property

        Public ReadOnly Property MediumURL() As String
            Get
                Return TCF_DIRECT_URL & TCPF_PHOTO_STRING & "?id=" & Me.ID & "&size=medium"
            End Get
        End Property

        Public ReadOnly Property SquareURL() As String
            Get
                Return TCF_DIRECT_URL & TCPF_PHOTO_STRING & "?id=" & Me.ID & "&size=square"
            End Get
        End Property

        Public ReadOnly Property TileURL() As String
            Get
                Return TCF_DIRECT_URL & TCPF_PHOTO_STRING & "?id=" & Me.ID & "&size=tile"
            End Get
        End Property

        Public ReadOnly Property LargestURL() As String
            Get
                Return TCF_DIRECT_URL & TCPF_PHOTO_STRING & "?id=" & Me.ID & "&size=largest_available"
            End Get
        End Property

    End Class

End Namespace

File: Assemblies/Protean.Tools_Net7/Integration/Twitter/TwitterVB2/Objects/ATwiCliPhotoFile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Protean.Tools.Integration.Twitter.TwitterVB2
{
    public partial class ATwiCliPhotoFile : ATwiCliFile
    {

        #region TCPF Constants
        private const string TCPF_PHOTO_STRING = "photo.jpg";
        private const string TCPF_PHOTO_EXTENSION = ".jpg";
        #endregion
        private int m_iUserTags;
        private double m_dLongitude;
        private double m_dLatitude;
        private bool m_bShowMap;
        private string m_strCameraMake;
        private string m_strCameraModel;
        public int UserTags_Count
        {
            get
            {
                return m_iUserTags;

            }
            set
            {
                m_iUserTags = value;

            }
        }

        public double Longitude
        {
            get
            {
                return m_dLongitude;

            }
            set
            {
                m_dLongitude = value;
            }
        }

        public double Latitude
        {
            get
            {
                return m_dLatitude;

            }
            set
            {
                m_dLatitude = value;

            }
        }

        public bool ShowMap
        {
            get
            {
                return m_bShowMap;

            }
            set
            {
                m_bShowMap = value;

            }
        }

        public string CameraMake
        {
            get
            {
                return m_strCameraMake;

            }
            set
            {
                m_strCameraMake = value;

            }
        }

        public string CameraModel
        {
            get
            {
                return m_strCameraModel;

            }
            set
            {
                m_strCameraModel = value;

            }
        }

        public override string GetDirectURL()
        {
            string GetDirectURLRet = default;
            GetDirectURLRet = TCF_DIRECT_URL + TCPF_PHOTO_STRING + "?id=" + this.ID + "&size=original";
            return GetDirectURLRet;
        }

        public override string GetExtension()
        {
            return TCPF_PHOTO_EXTENSION;

        }

        public string LargeURL
        {
            get
            {
                return TCF_DIRECT_URL + TCPF_PHOTO_STRING + "?id=" + this.ID + "&size=large";
            }
        }

        public string SmallURL
        {
            get
            {
                return TCF_DIRECT_URL + TCPF_PHOTO_STRING + "?id=" + this.ID + "&size=small";
            }
        }

        public string MediumURL
        {
            get
            {
                return TCF_DIRECT_URL + TCPF_PHOTO_STRING + "?id=" + this.ID + "&size=medium";
            }
        }

        public string SquareURL
        {
            get
            {
                return TCF_DIRECT_URL + TCPF_PHOTO_STRING + "?id=" + this.ID + "&size=square";
            }
        }

        public string TileURL
        {
            get
            {
                return TCF_DIRECT_URL + TCPF_PHOTO_STRING + "?id=" + this.ID + "&size=tile";
            }
        }

        public string LargestURL
        {
            get
            {
                return TCF_DIRECT_URL + TCPF_PHOTO_STRING + "?id=" + this.ID + "&size=largest_available";
            }
        }

    }

}
